gtp2ogs.js

```diff
@@ -2225,10 +2423,21 @@ function num2gtpchar(num) { /* {{{ */
         return ".";
     return "abcdefghjklmnopqrstuvwxyz"[num];
 } /* }}} */
-function rank2str(r) { /* {{{ */
-    r = r.toFixed();
-    if (r >= 30)  return (r-30+1) + 'd';
-    else          return (30-r) + 'k';
+function rankToString(r) { /* {{{ */
+    r = Math.floor(r);
+    if (r >= 30)  return (r-30+1) + 'd'; // r>=30 : 1 dan or stronger
+    else          return (30-r) + 'k'; // r<30 : 1 kyu or weaker
+} /* }}} */
+function timespanToDisplayString(timespan) {
+    let ss = timespan % 60;
+    let mm = Math.floor(timespan / 60 % 60);
+    let hh = Math.floor(timespan / (60*60) % 24);
+    let dd = Math.floor(timespan / (60*60*24));
+    let text = ["days", "hours", "minutes", "seconds"];
+    return [dd, hh, mm, ss]
+    .map((e, i) => e === 0 ? "" : `${e} ${text[i]}`)
+    .filter(e => e !== "")
+    .join(" ");
```

Comment:
I would prefer 2-4 extra spaces at the start of line 2438-2440 to indicate the `.map.filter.join` lines are part of the previous line.

Reply:
not sure where to put them, so leaving as it is :100: